File: plugins/objc-kit/testdata/good/Ok.m
#import <Foundation/Foundation.h>
#import <os/log.h>

@interface Ok : NSObject
- (void)run;
@end

@implementation Ok
- (void)run {
  os_log_t log = os_log_create("farm", "ok");
  os_log_info(log, "ok path");
  // Documented intentional seam; keep allow on the smell line when needed.
  NSLog(@"documented"); // objc-rg-allow: fixture documents allow marker for intentional NSLog seam
}
@end

File: plugins/delphi-kit/testdata/good/ok.pas
unit Ok;

interface

implementation

type
  TThing = record
    X: Integer;
    Y: Integer;
  end;

procedure GoodPath;
var
  P: Pointer;
  T: TThing;
begin
  New(P); // typed path preferred in product; fixture uses New
  T.X := 1;
  T.Y := 2;
  // Documented intentional seam; keep allow on the smell line.
  WriteLn('fixture'); // delphi-rg-allow: fixture documents allow marker for intentional WriteLn seam
end;

end.
